jhotdraw-core/src/main/java/org/jhotdraw/draw/figure/BezierFigure.java

```diff
@@ -210,11 +210,14 @@ public boolean contains(Point2D.Double p, double scaleDenominator) {
         Point2D.Double p2 = cp.get(path.size() - 1, 0);
         // FIXME - Check here, if caps path contains the point
         if (Geom.lineContainsPoint(p1.x, p1.y, p2.x, p2.y, p.x, p.y, tolerance)) {
-          return true;
+          result = true;
         }
       }
+    } else {
```

Comment:
This else is useless and should be removed. A return was added when the original intent of the change was to have only one return statement in the method. Moreover, the value of the variable "result" is already at false when it reaches this point.